pyqtorch/analog.py

```diff
@@ -533,6 +535,9 @@ def __init__(
             GeneratorType.PARAMETRIC_OPERATION: self._parametric_generator,
         }
 
+        # to avoid recomputing hamiltonians and evolution
+        self._cache_hamiltonian_evo: dict[str, Tensor] = {}
```

Comment:
```suggestion
        self._cache_hamiltonian_evo: dict[str, Tensor] = dict()
```

I find it less ambiguous.